src/base/Atomics.hh

```diff
@@ -25,8 +25,12 @@ CELER_FORCEINLINE_FUNCTION T atomic_add(T* address, T value)
     return atomicAdd(address, value);
 #else
     CELER_EXPECT(address);
-    T initial = *address;
-    *address += value;
+    T initial;
+#    pragma omp atomic capture
+    {
+        initial = *address;
+        *address += value;
+    }
```

Comment:
Another solution to the flag propagation issue you pointed out would be to wrap this in an `#ifdef _OPENMP`.

Reply:
I ended up taking this approach so that we don't have to worry about `-Wno-unknown-pragmas` everywhere. Now we only use the OpenMP version of atomic_add when it's actually needed by virtue of appearing in an `omp parallel for` (the main celeritas library and the demo-loop library).